pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -66,6 +68,52 @@ type Image struct {
 	SecretRef *corev1.LocalObjectReference `json:"credentials,omitempty"`
 }
 
+// Runtime represents the runtime-image, created using parts of builder-image, and a different
+// base-image than originally.
+type Runtime struct {
+	// Base runtime base image.
+	// +optional
+	Base Image `json:"base,omitempty"`
+
+	// Env environment variables for runtime.
+	// +optional
+	Env map[string]string `json:"env,omitempty"`
+
+	// Labels map of additional labels to be applied on image.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// WorkDir runtime image working directory `WORKDIR`.
+	// +optional
+	WorkDir string `json:"workDir,omitempty"`
+
+	// Run arbitrary commands to run before copying data into runtime-image.
+	// +optional
+	Run []string `json:"run,omitempty"`
+
+	// Paths list of directories/files to be copied into runtime-image, using colon ":" to split up source and destination paths.
+	// +optional
+	Paths []string `json:"paths,omitempty"`
+
+	// User definitions of user and group for runtime-image.
+	User *User `json:"user,omitempty"`
+
+	// Entrypoint runtime-image entrypoint.
+	// +optional
+	Entrypoint []string `json:"entrypoint,omitempty"`
+}
+
+// UUser holds the user name and group information for runtime-image.
+type User struct {
+	// Name user name to be employed in runtime-image.
+	// +optional
+	Name string `json:"name,omitempty"`
```

Comment:
I think this cannot be optional. The `user` field within runtime needs to be optional (and it is, you also have a reference there which if I now understand it correctly means it can become `nil` in the go code. But, if the user is specified, then the name is required and only the group is optional. I think that one can also specify a UID here.

Reply:
> I think this cannot be optional. The `user` field within runtime needs to be optional (and it is, you also have a reference there which if I now understand it correctly means it can become `nil` in the go code. But, if the user is specified, then the name is required and only the group is optional.

Okay, you have a good point, thanks. In a nutshell, if the `.spec.runtime.user` is specified, then `.spec.runtime.user.name` must be informed. I'll change this.

> I think that one can also specify a UID here.

Yes, it's possible to use `UID` in this attribute. Which also brings me to the same point you had before, `.spec.runtime.user.name` sounds vague given where and how it can be employed.